### PR TITLE
Bugfix/290 in user profile picture need response 401 unauthorized resolve

### PR DESCRIPTION
**Pull request by NazarVavrushchak**

### CODE REVIEW

**Review by ZenukD — APPROVED** [review]

File: core/src/main/java/greencity/config/SecurityConfig.java [hunk 1]
```diff
@@ -130,10 +132,12 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                                 "/ownSecurity/signIn",
                                 "/ownSecurity/updatePassword")
                         .permitAll()
+                        .requestMatchers(HttpMethod.GET, "/user/isOnline/{userId}/")
+                        .authenticated()
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/config/SecurityConfig.java:136]: Why do do you need it to solve current task?
.requestMatchers(HttpMethod.GET, "/user/isOnline/{userId}/")
                        .authenticated()

File: core/src/main/java/greencity/controller/UserController.java [hunk 2]
```diff
@@ -59,20 +66,26 @@ public class UserController {
...
+        if (principal == null) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+        try {
+            UserStatusDto updatedStatus = userService.updateStatus(
+                    userStatusDto.getId(), userStatusDto.getUserStatus(), principal.getName());
+            return ResponseEntity.status(HttpStatus.OK).body(updatedStatus);
+        } catch (BadUpdateRequestException | LowRoleLevelException e) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: Why did You modify this method? How does it help to solve task 290?

File: core/src/main/java/greencity/controller/UserController.java [hunk 3]
```diff
@@ -244,35 +257,36 @@ public ResponseEntity<UserUpdateDto> updateUser(@Valid @RequestBody UserUpdateDt
...
-            .status(HttpStatus.OK)
-            .body(userService.getAvailableCustomShoppingListItems(userId, habitId));
+                .status(HttpStatus.OK)
+                .body(userService.getAvailableCustomShoppingListItems(userId, habitId));
     }
 
     /**
-     * Counts all users by user {@link UserStatus} ACTIVATED.
+     * Counts all users by user {@link UserStatus} ACTIVATEDd
      *
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: I guess You have added letter 'd' accidently.

File: core/src/main/java/greencity/controller/UserController.java [hunk 4]
```diff
@@ -368,16 +385,19 @@ public ResponseEntity<UserProfileDtoResponse> getUserProfileInformation(
...
-        return ResponseEntity
-            .status(HttpStatus.OK)
-            .body(userService.checkIfTheUserIsOnline(userId));
+            @Parameter(description = "Id of the user. Cannot be empty.") @PathVariable Long userId) {
+        try {
+            boolean isOnline = userService.checkIfTheUserIsOnline(userId);
+            return ResponseEntity.status(HttpStatus.OK).body(isOnline);
+        } catch (WrongIdException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(false);
+        }
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: Why did You modify this method? How does it help to solve task 290?
You have already fixed it in the different PR.

File: core/src/main/java/greencity/controller/UserController.java [hunk 5]
```diff
@@ -388,15 +408,20 @@ public ResponseEntity<Boolean> checkIfTheUserIsOnline(
...
+            @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN)
     })
     @GetMapping("/{userId}/profileStatistics/")
     public ResponseEntity<UserProfileStatisticsDto> getUserProfileStatistics(
-        @Parameter(description = "Id of current user. Cannot be empty.") @PathVariable @CurrentUserId Long userId) {
+            @PathVariable Long userId
+            , @Parameter(description = "Id of current user. Cannot be empty.") @CurrentUserId Long currentUserId) {
+        if (!currentUserId.equals(userId)) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
```
Comment (VitaliiChernenko) [inline core/src/main/java/greencity/controller/UserController.java]: Why did You modify this method? How does it help to solve task 290?
You have already done it in the different PR.

**Review by DmytroLysenko1 — APPROVED** [review]
